pkg/controller/dns/util.go

```diff
@@ -45,3 +52,27 @@ func extractResourceTypes(resourceTypes []string) map[manifests.ResourceType]str
 
 	return ret
 }
+
+func generateManifestsConf(config *config.Config, obj ExternalDNSCRDConfiguration) (*manifests.ExternalDnsConfig, error) {
+	inputDNSConf := buildInputDNSConfig(obj)
+	manifestsConf, err := manifests.NewExternalDNSConfig(config, inputDNSConf)
+	if err != nil {
+		return nil, util.NewUserError(err, "failed to generate ExternalDNS resources: "+err.Error())
+	}
+
+	return manifestsConf, nil
+}
+
+func deployExternalDNSResources(ctx context.Context, client client.Client, manifestsConf *manifests.ExternalDnsConfig, owners []metav1.OwnerReference) error {
+	// create the ExternalDNS resources
+	multiError := &multierror.Error{}
+
+	for _, resource := range manifestsConf.Resources() {
+		resource.SetOwnerReferences(owners)
+
+		currentResourceErr := util.Upsert(ctx, client, resource)
+		multiError = multierror.Append(multiError, currentResourceErr)
```

Comment:
should only append if error != nil?

Reply:
nil errors are ignored, unless I'm misinterpreting the docs - https://pkg.go.dev/github.com/hashicorp/go-multierror@v1.1.1#Append